File: 2002/101 VB.NET Samples/VB.NET - Windows Forms - How-To Inherited Windows Forms/frmBase.vb
'Copyright (C) 2002 Microsoft Corporation
'All rights reserved.
'THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER 
'EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES OF 
'MERCHANTIBILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.

'Requires the Trial or Release version of Visual Studio .NET Professional (or greater).

Option Strict On

Imports System.Data.SqlClient

Public Class frmBase
    Inherits System.Windows.Forms.Form

    ' Initialize constants for connecting to the database
    ' and displaying a connection error to the user.
    Protected Const SQL_CONNECTION_STRING As String = _
        "Server=localhost;" & _
        "DataBase=Northwind;" & _
        "Integrated Security=SSPI"

    Protected Const MSDE_CONNECTION_STRING As String = _
        "Server=(local)\NetSDK;" & _
        "DataBase=Northwind;" & _
        "Integrated Security=SSPI"

    Protected Const CONNECTION_ERROR_MSG As String = _
        "To run this sample, you must have SQL " & _
        "or MSDE with the Northwind database installed.  For " & _
        "instructions on installing MSDE, view the ReadMe file."

    Protected DidPreviouslyConnect As Boolean = False
    Protected DidCreateTable As Boolean = False
    Protected connectionString As String = SQL_CONNECTION_STRING

#Region " Windows Form Designer generated code "

    Public Sub New()
        MyBase.New()

        'This call is required by the Windows Form Designer.
        InitializeComponent()

        'Add any initialization after the InitializeComponent() call

    End Sub

    'Form overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
        If disposing Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub

    'Required by the Windows Form Designer
    Private components As System.ComponentModel.IContainer

    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.  
    'Do not modify it using the code editor.
    Friend WithEvents lblTitle As System.Windows.Forms.Label
    Friend WithEvents btnClose As System.Windows.Forms.Button
    Friend WithEvents PictureBox1 As System.Windows.Forms.PictureBox
    Protected WithEvents lblProtected As System.Windows.Forms.Label
    Private WithEvents lblPrivate As System.Windows.Forms.Label
    <System.Diagnostics.DebuggerStepThrough()> Private Sub InitializeComponent()
        Dim resources As System.Resources.ResourceManager = New System.Resources.ResourceManager(GetType(frmBase))
        Me.lblTitle = New System.Windows.Forms.Label()
        Me.btnClose = New System.Windows.Forms.Button()
        Me.PictureBox1 = New System.Windows.Forms.PictureBox()
        Me.lblProtected = New System.Windows.Forms.Label()
        Me.lblPrivate = New System.Windows.Forms.Label()
        Me.SuspendLayout()
        '
        'lblTitle
        '
        Me.lblTitle.AccessibleDescription = resources.GetString("lblTitle.AccessibleDescription")
        Me.lblTitle.AccessibleName = resources.GetString("lblTitle.AccessibleName")
        Me.lblTitle.Anchor = CType(resources.GetObject("lblTitle.Anchor"), System.Windows.Forms.AnchorStyles)
        Me.lblTitle.AutoSize = CType(resources.GetObject("lblTitle.AutoSize"), Boolean)
        Me.lblTitle.BackColor = System.Drawing.SystemColors.Control
        Me.lblTitle.Dock = CType(resources.GetObject("lblTitle.Dock"), System.Windows.Forms.DockStyle)
        Me.lblTitle.Enabled = CType(resources.GetObject("lblTitle.Enabled"), Boolean)
        Me.lblTitle.Font = CType(resources.GetObject("lblTitle.Font"), System.Drawing.Font)
        Me.lblTitle.ForeColor = System.Drawing.SystemColors.ControlText
        Me.lblTitle.Image = CType(resources.GetObject("lblTitle.Image"), System.Drawing.Image)
        Me.lblTitle.ImageAlign = CType(resources.GetObject("lblTitle.ImageAlign"), System.Drawing.ContentAlignment)
        Me.lblTitle.ImageIndex = CType(resources.GetObject("lblTitle.ImageIndex"), Integer)
        Me.lblTitle.ImeMode = CType(resources.GetObject("lblTitle.ImeMode"), System.Windows.Forms.ImeMode)
        Me.lblTitle.Location = CType(resources.GetObject("lblTitle.Location"), System.Drawing.Point)
        Me.lblTitle.Name = "lblTitle"
        Me.lblTitle.RightToLeft = CType(resources.GetObject("lblTitle.RightToLeft"), System.Windows.Forms.RightToLeft)
        Me.lblTitle.Size = CType(resources.GetObject("lblTitle.Size"), System.Drawing.Size)
        Me.lblTitle.TabIndex = CType(resources.GetObject("lblTitle.TabIndex"), Integer)
        Me.lblTitle.Text = resources.GetString("lblTitle.Text")
        Me.lblTitle.TextAlign = CType(resources.GetObject("lblTitle.TextAlign"), System.Drawing.ContentAlignment)
        Me.lblTitle.Visible = CType(resources.GetObject("lblTitle.Visible"), Boolean)
        '
        'btnClose
        '
        Me.btnClose.AccessibleDescription = resources.GetString("btnClose.AccessibleDescription")
        Me.btnClose.AccessibleName = resources.GetString("btnClose.AccessibleName")
        Me.btnClose.Anchor = CType(resources.GetObject("btnClose.Anchor"), System.Windows.Forms.AnchorStyles)
        Me.btnClose.BackColor = System.Drawing.SystemColors.Control
        Me.btnClose.BackgroundImage = CType(resources.GetObject("btnClose.BackgroundImage"), System.Drawing.Image)
        Me.btnClose.Dock = CType(resources.GetObject("btnClose.Dock"), System.Windows.Forms.DockStyle)
        Me.btnClose.Enabled = CType(resources.GetObject("btnClose.Enabled"), Boolean)
        Me.btnClose.FlatStyle = CType(resources.GetObject("btnClose.FlatStyle"), System.Windows.Forms.FlatStyle)
        Me.btnClose.Font = CType(resources.GetObject("btnClose.Font"), System.Drawing.Font)
        Me.btnClose.ForeColor = System.Drawing.SystemColors.ControlText
        Me.btnClose.Image = CType(resources.GetObject("btnClose.Image"), System.Drawing.Image)
        Me.btnClose.ImageAlign = CType(resources.GetObject("btnClose.ImageAlign"), System.Drawing.ContentAlignment)
        Me.btnClose.ImageIndex = CType(resources.GetObject("btnClose.ImageIndex"), Integer)
        Me.btnClose.ImeMode = CType(resources.GetObject("btnClose.ImeMode"), System.Windows.Forms.ImeMode)
        Me.btnClose.Location = CType(resources.GetObject("btnClose.Location"), System.Drawing.Point)
        Me.btnClose.Name = "btnClose"
        Me.btnClose.RightToLeft = CType(resources.GetObject("btnClose.RightToLeft"), System.Windows.Forms.RightToLeft)
        Me.btnClose.Size = CType(resources.GetObject("btnClose.Size"), System.Drawing.Size)
        Me.btnClose.TabIndex = CType(resources.GetObject("btnClose.TabIndex"), Integer)
        Me.btnClose.Text = resources.GetString("btnClose.Text")
        Me.btnClose.TextAlign = CType(resources.GetObject("btnClose.TextAlign"), System.Drawing.ContentAlignment)
        Me.btnClose.Visible = CType(resources.GetObject("btnClose.Visible"), Boolean)
        '
        'PictureBox1
        '
        Me.PictureBox1.AccessibleDescription = resources.GetString("PictureBox1.AccessibleDescription")
        Me.PictureBox1.AccessibleName = resources.GetString("PictureBox1.AccessibleName")
        Me.PictureBox1.Anchor = CType(resources.GetObject("PictureBox1.Anchor"), System.Windows.Forms.AnchorStyles)
        Me.PictureBox1.BackgroundImage = CType(resources.GetObject("PictureBox1.BackgroundImage"), System.Drawing.Image)
        Me.PictureBox1.Dock = CType(resources.GetObject("PictureBox1.Dock"), System.Windows.Forms.DockStyle)
        Me.PictureBox1.Enabled = CType(resources.GetObject("PictureBox1.Enabled"), Boolean)
        Me.PictureBox1.Font = CType(resources.GetObject("PictureBox1.Font"), System.Drawing.Font)
        Me.PictureBox1.Image = CType(resources.GetObject("PictureBox1.Image"), System.Drawing.Bitmap)
        Me.PictureBox1.ImeMode = CType(resources.GetObject("PictureBox1.ImeMode"), System.Windows.Forms.ImeMode)
        Me.PictureBox1.Location = CType(resources.GetObject("PictureBox1.Location"), System.Drawing.Point)
        Me.PictureBox1.Name = "PictureBox1"
        Me.PictureBox1.RightToLeft = CType(resources.GetObject("PictureBox1.RightToLeft"), System.Windows.Forms.RightToLeft)
        Me.PictureBox1.Size = CType(resources.GetObject("PictureBox1.Size"), System.Drawing.Size)
        Me.PictureBox1.SizeMode = CType(resources.GetObject("PictureBox1.SizeMode"), System.Windows.Forms.PictureBoxSizeMode)
        Me.PictureBox1.TabIndex = CType(resources.GetObject("PictureBox1.TabIndex"), Integer)
        Me.PictureBox1.TabStop = False
        Me.PictureBox1.Text = resources.GetString("PictureBox1.Text")
        Me.PictureBox1.Visible = CType(resources.GetObject("PictureBox1.Visible"), Boolean)
        '
        'lblProtected
        '
        Me.lblProtected.AccessibleDescription = resources.GetString("lblProtected.AccessibleDescription")
        Me.lblProtected.AccessibleName = resources.GetString("lblProtected.AccessibleName")
        Me.lblProtected.Anchor = CType(resources.GetObject("lblProtected.Anchor"), System.Windows.Forms.AnchorStyles)
        Me.lblProtected.AutoSize = CType(resources.GetObject("lblProtected.AutoSize"), Boolean)
        Me.lblProtected.Dock = CType(resources.GetObject("lblProtected.Dock"), System.Windows.Forms.DockStyle)
        Me.lblProtected.Enabled = CType(resources.GetObject("lblProtected.Enabled"), Boolean)
        Me.lblProtected.Font = CType(resources.GetObject("lblProtected.Font"), System.Drawing.Font)
        Me.lblProtected.Image = CType(resources.GetObject("lblProtected.Image"), System.Drawing.Image)
        Me.lblProtected.ImageAlign = CType(resources.GetObject("lblProtected.ImageAlign"), System.Drawing.ContentAlignment)
        Me.lblProtected.ImageIndex = CType(resources.GetObject("lblProtected.ImageIndex"), Integer)
        Me.lblProtected.ImeMode = CType(resources.GetObject("lblProtected.ImeMode"), System.Windows.Forms.ImeMode)
        Me.lblProtected.Location = CType(resources.GetObject("lblProtected.Location"), System.Drawing.Point)
        Me.lblProtected.Name = "lblProtected"
        Me.lblProtected.RightToLeft = CType(resources.GetObject("lblProtected.RightToLeft"), System.Windows.Forms.RightToLeft)
        Me.lblProtected.Size = CType(resources.GetObject("lblProtected.Size"), System.Drawing.Size)
        Me.lblProtected.TabIndex = CType(resources.GetObject("lblProtected.TabIndex"), Integer)
        Me.lblProtected.Text = resources.GetString("lblProtected.Text")
        Me.lblProtected.TextAlign = CType(resources.GetObject("lblProtected.TextAlign"), System.Drawing.ContentAlignment)
        Me.lblProtected.Visible = CType(resources.GetObject("lblProtected.Visible"), Boolean)
        '
        'lblPrivate
        '
        Me.lblPrivate.AccessibleDescription = resources.GetString("lblPrivate.AccessibleDescription")
        Me.lblPrivate.AccessibleName = resources.GetString("lblPrivate.AccessibleName")
        Me.lblPrivate.Anchor = CType(resources.GetObject("lblPrivate.Anchor"), System.Windows.Forms.AnchorStyles)
        Me.lblPrivate.AutoSize = CType(resources.GetObject("lblPrivate.AutoSize"), Boolean)
        Me.lblPrivate.Dock = CType(resources.GetObject("lblPrivate.Dock"), System.Windows.Forms.DockStyle)
        Me.lblPrivate.Enabled = CType(resources.GetObject("lblPrivate.Enabled"), Boolean)
        Me.lblPrivate.Font = CType(resources.GetObject("lblPrivate.Font"), System.Drawing.Font)
        Me.lblPrivate.Image = CType(resources.GetObject("lblPrivate.Image"), System.Drawing.Image)
        Me.lblPrivate.ImageAlign = CType(resources.GetObject("lblPrivate.ImageAlign"), System.Drawing.ContentAlignment)
        Me.lblPrivate.ImageIndex = CType(resources.GetObject("lblPrivate.ImageIndex"), Integer)
        Me.lblPrivate.ImeMode = CType(resources.GetObject("lblPrivate.ImeMode"), System.Windows.Forms.ImeMode)
        Me.lblPrivate.Location = CType(resources.GetObject("lblPrivate.Location"), System.Drawing.Point)
        Me.lblPrivate.Name = "lblPrivate"
        Me.lblPrivate.RightToLeft = CType(resources.GetObject("lblPrivate.RightToLeft"), System.Windows.Forms.RightToLeft)
        Me.lblPrivate.Size = CType(resources.GetObject("lblPrivate.Size"), System.Drawing.Size)
        Me.lblPrivate.TabIndex = CType(resources.GetObject("lblPrivate.TabIndex"), Integer)
        Me.lblPrivate.Text = resources.GetString("lblPrivate.Text")
        Me.lblPrivate.TextAlign = CType(resources.GetObject("lblPrivate.TextAlign"), System.Drawing.ContentAlignment)
        Me.lblPrivate.Visible = CType(resources.GetObject("lblPrivate.Visible"), Boolean)
        '
        'frmBase
        '
        Me.AccessibleDescription = CType(resources.GetObject("$this.AccessibleDescription"), String)
        Me.AccessibleName = CType(resources.GetObject("$this.AccessibleName"), String)
        Me.Anchor = CType(resources.GetObject("$this.Anchor"), System.Windows.Forms.AnchorStyles)
        Me.AutoScaleBaseSize = CType(resources.GetObject("$this.AutoScaleBaseSize"), System.Drawing.Size)
        Me.AutoScroll = CType(resources.GetObject("$this.AutoScroll"), Boolean)
        Me.AutoScrollMargin = CType(resources.GetObject("$this.AutoScrollMargin"), System.Drawing.Size)
        Me.AutoScrollMinSize = CType(resources.GetObject("$this.AutoScrollMinSize"), System.Drawing.Size)
        Me.BackgroundImage = CType(resources.GetObject("$this.BackgroundImage"), System.Drawing.Image)
        Me.ClientSize = CType(resources.GetObject("$this.ClientSize"), System.Drawing.Size)
        Me.Controls.AddRange(New System.Windows.Forms.Control() {Me.lblPrivate, Me.lblProtected, Me.PictureBox1, Me.btnClose, Me.lblTitle})
        Me.Dock = CType(resources.GetObject("$this.Dock"), System.Windows.Forms.DockStyle)
        Me.Enabled = CType(resources.GetObject("$this.Enabled"), Boolean)
        Me.Font = CType(resources.GetObject("$this.Font"), System.Drawing.Font)
        Me.Icon = CType(resources.GetObject("$this.Icon"), System.Drawing.Icon)
        Me.ImeMode = CType(resources.GetObject("$this.ImeMode"), System.Windows.Forms.ImeMode)
        Me.Location = CType(resources.GetObject("$this.Location"), System.Drawing.Point)
        Me.MaximizeBox = False
        Me.MaximumSize = CType(resources.GetObject("$this.MaximumSize"), System.Drawing.Size)
        Me.MinimumSize = CType(resources.GetObject("$this.MinimumSize"), System.Drawing.Size)
        Me.Name = "frmBase"
        Me.RightToLeft = CType(resources.GetObject("$this.RightToLeft"), System.Windows.Forms.RightToLeft)
        Me.StartPosition = CType(resources.GetObject("$this.StartPosition"), System.Windows.Forms.FormStartPosition)
        Me.Text = resources.GetString("$this.Text")
        Me.Visible = CType(resources.GetObject("$this.Visible"), Boolean)
        Me.ResumeLayout(False)

    End Sub

#End Region

    ' Creates a connection to the database and uses a SqlDataAdapter
    ' object to fill a DataSet.
    Protected Function GetDataSource() As DataSet

        Static ConnectionString As String = SQL_CONNECTION_STRING
        Static DidPreviouslyConnect As Boolean = False

        ' Display a status message saying that we're attempting to connect.
        ' This only needs to be done the very first time a connection is
        ' attempted.  After we've determined that MSDE or SQL Server is
        ' installed, this message no longer needs to be displayed.
        Dim frmStatusMessage As New frmStatus()
        If Not DidPreviouslyConnect Then
            frmStatusMessage.Show("Connecting to SQL Server")
        End If

        ' Attempt to connect to the local SQL server instance, and a local
        ' MSDE installation (with Northwind).  
        Dim IsConnecting As Boolean = True
        While IsConnecting

            Try
                ' The SqlConnection class allows you to communicate with SQL Server.
                ' The constructor accepts a connection string as an argument.  This
                ' connection string uses Integrated Security, which means that you 
                ' must have a login in SQL Server, or be part of the Administrators
                ' group for this to work.
                Dim cnNorthwind As New SqlConnection(ConnectionString)

                ' The SqlDataAdapter is used to move data between SQL Server, 
                ' and a DataSet.
                Dim da As New SqlDataAdapter( _
                    "SELECT ProductID, ProductName, UnitPrice, UnitsInStock " & _
                    "FROM products", cnNorthwind)

                Dim dsProducts As New DataSet()

                ' Populate the DataSet with the information from the products
                ' table.  Since a DataSet can hold multiple result sets, it's
                ' a good idea to "name" the result set when you populate the
                ' DataSet.  In this case, the result set is named "Products".
                da.Fill(dsProducts, "Products")

                ' Data has been successfully retrieved, so break out of the loop
                ' and close the status form.
                IsConnecting = False
                DidPreviouslyConnect = True
                frmStatusMessage.Close()

                Return dsProducts

            Catch exc As Exception
                If ConnectionString = SQL_CONNECTION_STRING Then
                    ' Couldn't connect to SQL Server.  Now try MSDE.
                    ConnectionString = MSDE_CONNECTION_STRING
                    frmStatusMessage.Show("Connecting to MSDE")
                Else
                    ' Unable to connect to SQL Server or MSDE
                    frmStatusMessage.Close()
                    MessageBox.Show("To run this sample, you must have SQL " & _
                        "or MSDE with the Northwind database installed.  For " & _
                        "instructions on installing MSDE, view the ReadMe file.", _
                        "Connection Problem", MessageBoxButtons.OK, _
                        MessageBoxIcon.Error)
                    End
                End If
            End Try
        End While

        frmStatusMessage.Close()

    End Function

    Private Sub frmBase_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

    End Sub
End Class

File: 2003/CS/VisualInheritance/frmBase.cs
//Copyright (C) 2002 Microsoft Corporation
//All rights reserved.
//THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER 
//EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED WARRANTIES OF 
//MERCHANTIBILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
//Requires the Trial or Release version of Visual Studio .NET Professional (or greater).

using System;
using System.Windows.Forms;
using System.Data.SqlClient;
using System.Data;

public class frmBase: System.Windows.Forms.Form {

    // Initialize constants for connecting to the database
    // and displaying a connection error to the user.

    protected const string SQL_CONNECTION_STRING = "Server=localhost;" + 
		"DataBase=Northwind;Integrated Security=SSPI";

    protected const string MSDE_CONNECTION_STRING = @"Server=(local)\NetSDK;" +
        "DataBase=Northwind;Integrated Security=SSPI";

    protected const string CONNECTION_ERROR_MSG = "To run this sample, you must have SQL " +
		"or MSDE with the Northwind database installed.  For " + 
        "instructions on installing MSDE, view the ReadMe file.";

    static bool DidPreviouslyConnect = false;
    protected bool DidCreateTable = false;
    static string connectionstring = SQL_CONNECTION_STRING;

#region " Windows Form Designer generated code "

    public frmBase () {

        //This call is required by the Windows Form Designer.

        InitializeComponent();

        //Add any initialization after the InitializeComponent() call

    }

    //Form overrides dispose to clean up the component list.

    protected override void Dispose(bool disposing) {

        if (disposing) {

            if (components != null) {

                components.Dispose();

            }

        }

        base.Dispose(disposing);

    }

    //Required by the Windows Form Designer

    private System.ComponentModel.IContainer components = null;

    //NOTE: The following procedure is required by the Windows Form Designer

    //It can be modified using the Windows Form Designer.  

    //Do not modify it using the code editor.

    protected System.Windows.Forms.Label lblTitle;

    protected System.Windows.Forms.Button btnClose;

    private System.Windows.Forms.PictureBox PictureBox1;

    protected System.Windows.Forms.Label lblProtected;

    private System.Windows.Forms.Label lblPrivate;

    private void InitializeComponent() {

        System.Resources.ResourceManager resources = new System.Resources.ResourceManager(typeof(frmBase));

        this.lblTitle = new System.Windows.Forms.Label();

        this.btnClose = new System.Windows.Forms.Button();

        this.PictureBox1 = new System.Windows.Forms.PictureBox();

        this.lblProtected = new System.Windows.Forms.Label();

        this.lblPrivate = new System.Windows.Forms.Label();

        this.SuspendLayout();

        //

        //lblTitle

        //

        this.lblTitle.AccessibleDescription = resources.GetString("lblTitle.AccessibleDescription");

        this.lblTitle.AccessibleName = resources.GetString("lblTitle.AccessibleName");

        this.lblTitle.Anchor = (System.Windows.Forms.AnchorStyles) resources.GetObject("lblTitle.Anchor");

        this.lblTitle.AutoSize = (bool) resources.GetObject("lblTitle.AutoSize");

        this.lblTitle.BackColor = System.Drawing.SystemColors.Control;

        this.lblTitle.Dock = (System.Windows.Forms.DockStyle) resources.GetObject("lblTitle.Dock");

        this.lblTitle.Enabled = (bool) resources.GetObject("lblTitle.Enabled");

        this.lblTitle.Font = (System.Drawing.Font) resources.GetObject("lblTitle.Font");

        this.lblTitle.ForeColor = System.Drawing.SystemColors.ControlText;

        this.lblTitle.Image = (System.Drawing.Image) resources.GetObject("lblTitle.Image");

        this.lblTitle.ImageAlign = (System.Drawing.ContentAlignment) resources.GetObject("lblTitle.ImageAlign");

        this.lblTitle.ImageIndex = (int) resources.GetObject("lblTitle.ImageIndex");

        this.lblTitle.ImeMode = (System.Windows.Forms.ImeMode) resources.GetObject("lblTitle.ImeMode");

        this.lblTitle.Location = (System.Drawing.Point) resources.GetObject("lblTitle.Location");

        this.lblTitle.Name = "lblTitle";

        this.lblTitle.RightToLeft = (System.Windows.Forms.RightToLeft) resources.GetObject("lblTitle.RightToLeft");

        this.lblTitle.Size = (System.Drawing.Size) resources.GetObject("lblTitle.Size");

        this.lblTitle.TabIndex = (int) resources.GetObject("lblTitle.TabIndex");

        this.lblTitle.Text = resources.GetString("lblTitle.Text");

        this.lblTitle.TextAlign = (System.Drawing.ContentAlignment) resources.GetObject("lblTitle.TextAlign");

        this.lblTitle.Visible = (bool) resources.GetObject("lblTitle.Visible");

        //

        //btnClose

        //

        this.btnClose.AccessibleDescription = resources.GetString("btnClose.AccessibleDescription");

        this.btnClose.AccessibleName = resources.GetString("btnClose.AccessibleName");

        this.btnClose.Anchor = (System.Windows.Forms.AnchorStyles) resources.GetObject("btnClose.Anchor");

        this.btnClose.BackColor = System.Drawing.SystemColors.Control;

        this.btnClose.BackgroundImage = (System.Drawing.Image) resources.GetObject("btnClose.BackgroundImage");

        this.btnClose.Dock = (System.Windows.Forms.DockStyle) resources.GetObject("btnClose.Dock");

        this.btnClose.Enabled = (bool) resources.GetObject("btnClose.Enabled");

        this.btnClose.FlatStyle = (System.Windows.Forms.FlatStyle) resources.GetObject("btnClose.FlatStyle");

        this.btnClose.Font = (System.Drawing.Font) resources.GetObject("btnClose.Font");

        this.btnClose.ForeColor = System.Drawing.SystemColors.ControlText;

        this.btnClose.Image = (System.Drawing.Image) resources.GetObject("btnClose.Image");

        this.btnClose.ImageAlign = (System.Drawing.ContentAlignment) resources.GetObject("btnClose.ImageAlign");

        this.btnClose.ImageIndex = (int) resources.GetObject("btnClose.ImageIndex");

        this.btnClose.ImeMode = (System.Windows.Forms.ImeMode) resources.GetObject("btnClose.ImeMode");

        this.btnClose.Location = (System.Drawing.Point) resources.GetObject("btnClose.Location");

        this.btnClose.Name = "btnClose";

        this.btnClose.RightToLeft = (System.Windows.Forms.RightToLeft) resources.GetObject("btnClose.RightToLeft");

        this.btnClose.Size = (System.Drawing.Size) resources.GetObject("btnClose.Size");

        this.btnClose.TabIndex = (int) resources.GetObject("btnClose.TabIndex");

        this.btnClose.Text = resources.GetString("btnClose.Text");

        this.btnClose.TextAlign = (System.Drawing.ContentAlignment) resources.GetObject("btnClose.TextAlign");

        this.btnClose.Visible = (bool) resources.GetObject("btnClose.Visible");

        //

        //PictureBox1

        //

        this.PictureBox1.AccessibleDescription = resources.GetString("PictureBox1.AccessibleDescription");

        this.PictureBox1.AccessibleName = resources.GetString("PictureBox1.AccessibleName");

        this.PictureBox1.Anchor = (System.Windows.Forms.AnchorStyles) resources.GetObject("PictureBox1.Anchor");

        this.PictureBox1.BackgroundImage = (System.Drawing.Image) resources.GetObject("PictureBox1.BackgroundImage");

        this.PictureBox1.Dock = (System.Windows.Forms.DockStyle) resources.GetObject("PictureBox1.Dock");

        this.PictureBox1.Enabled = (bool) resources.GetObject("PictureBox1.Enabled");

        this.PictureBox1.Font = (System.Drawing.Font) resources.GetObject("PictureBox1.Font");

        this.PictureBox1.Image = (System.Drawing.Bitmap) resources.GetObject("PictureBox1.Image");

        this.PictureBox1.ImeMode = (System.Windows.Forms.ImeMode) resources.GetObject("PictureBox1.ImeMode");

        this.PictureBox1.Location = (System.Drawing.Point) resources.GetObject("PictureBox1.Location");

        this.PictureBox1.Name = "PictureBox1";

        this.PictureBox1.RightToLeft = (System.Windows.Forms.RightToLeft) resources.GetObject("PictureBox1.RightToLeft");

        this.PictureBox1.Size = (System.Drawing.Size) resources.GetObject("PictureBox1.Size");

        this.PictureBox1.SizeMode = (System.Windows.Forms.PictureBoxSizeMode) resources.GetObject("PictureBox1.SizeMode");

        this.PictureBox1.TabIndex = (int) resources.GetObject("PictureBox1.TabIndex");

        this.PictureBox1.TabStop = false;

        this.PictureBox1.Text = resources.GetString("PictureBox1.Text");

        this.PictureBox1.Visible = (bool) resources.GetObject("PictureBox1.Visible");

        //

        //lblProtected

        //

        this.lblProtected.AccessibleDescription = resources.GetString("lblProtected.AccessibleDescription");

        this.lblProtected.AccessibleName = resources.GetString("lblProtected.AccessibleName");

        this.lblProtected.Anchor = (System.Windows.Forms.AnchorStyles) resources.GetObject("lblProtected.Anchor");

        this.lblProtected.AutoSize = (bool) resources.GetObject("lblProtected.AutoSize");

        this.lblProtected.Dock = (System.Windows.Forms.DockStyle) resources.GetObject("lblProtected.Dock");

        this.lblProtected.Enabled = (bool) resources.GetObject("lblProtected.Enabled");

        this.lblProtected.Font = (System.Drawing.Font) resources.GetObject("lblProtected.Font");

        this.lblProtected.Image = (System.Drawing.Image) resources.GetObject("lblProtected.Image");

        this.lblProtected.ImageAlign = (System.Drawing.ContentAlignment) resources.GetObject("lblProtected.ImageAlign");

        this.lblProtected.ImageIndex = (int) resources.GetObject("lblProtected.ImageIndex");

        this.lblProtected.ImeMode = (System.Windows.Forms.ImeMode) resources.GetObject("lblProtected.ImeMode");

        this.lblProtected.Location = (System.Drawing.Point) resources.GetObject("lblProtected.Location");

        this.lblProtected.Name = "lblProtected";

        this.lblProtected.RightToLeft = (System.Windows.Forms.RightToLeft) resources.GetObject("lblProtected.RightToLeft");

        this.lblProtected.Size = (System.Drawing.Size) resources.GetObject("lblProtected.Size");

        this.lblProtected.TabIndex = (int) resources.GetObject("lblProtected.TabIndex");

        this.lblProtected.Text = resources.GetString("lblProtected.Text");

        this.lblProtected.TextAlign = (System.Drawing.ContentAlignment) resources.GetObject("lblProtected.TextAlign");

        this.lblProtected.Visible = (bool) resources.GetObject("lblProtected.Visible");

        //

        //lblPrivate

        //

        this.lblPrivate.AccessibleDescription = resources.GetString("lblPrivate.AccessibleDescription");

        this.lblPrivate.AccessibleName = resources.GetString("lblPrivate.AccessibleName");

        this.lblPrivate.Anchor = (System.Windows.Forms.AnchorStyles) resources.GetObject("lblPrivate.Anchor");

        this.lblPrivate.AutoSize = (bool) resources.GetObject("lblPrivate.AutoSize");

        this.lblPrivate.Dock = (System.Windows.Forms.DockStyle) resources.GetObject("lblPrivate.Dock");

        this.lblPrivate.Enabled = (bool) resources.GetObject("lblPrivate.Enabled");

        this.lblPrivate.Font = (System.Drawing.Font) resources.GetObject("lblPrivate.Font");

        this.lblPrivate.Image = (System.Drawing.Image) resources.GetObject("lblPrivate.Image");

        this.lblPrivate.ImageAlign = (System.Drawing.ContentAlignment) resources.GetObject("lblPrivate.ImageAlign");

        this.lblPrivate.ImageIndex = (int) resources.GetObject("lblPrivate.ImageIndex");

        this.lblPrivate.ImeMode = (System.Windows.Forms.ImeMode) resources.GetObject("lblPrivate.ImeMode");

        this.lblPrivate.Location = (System.Drawing.Point) resources.GetObject("lblPrivate.Location");

        this.lblPrivate.Name = "lblPrivate";

        this.lblPrivate.RightToLeft = (System.Windows.Forms.RightToLeft) resources.GetObject("lblPrivate.RightToLeft");

        this.lblPrivate.Size = (System.Drawing.Size) resources.GetObject("lblPrivate.Size");

        this.lblPrivate.TabIndex = (int) resources.GetObject("lblPrivate.TabIndex");

        this.lblPrivate.Text = resources.GetString("lblPrivate.Text");

        this.lblPrivate.TextAlign = (System.Drawing.ContentAlignment) resources.GetObject("lblPrivate.TextAlign");

        this.lblPrivate.Visible = (bool) resources.GetObject("lblPrivate.Visible");

        //

        //frmBase

        //

        this.AccessibleDescription = (string) resources.GetObject("$this.AccessibleDescription");

        this.AccessibleName = (string) resources.GetObject("$this.AccessibleName");

        this.Anchor = (System.Windows.Forms.AnchorStyles) resources.GetObject("$this.Anchor");

        this.AutoScaleBaseSize = (System.Drawing.Size) resources.GetObject("$this.AutoScaleBaseSize");

        this.AutoScroll = (bool) resources.GetObject("$this.AutoScroll");

        this.AutoScrollMargin = (System.Drawing.Size) resources.GetObject("$this.AutoScrollMargin");

        this.AutoScrollMinSize = (System.Drawing.Size) resources.GetObject("$this.AutoScrollMinSize");

        this.BackgroundImage = (System.Drawing.Image) resources.GetObject("$this.BackgroundImage");

        this.ClientSize = (System.Drawing.Size) resources.GetObject("$this.ClientSize");

        this.Controls.AddRange(new System.Windows.Forms.Control[] {this.lblPrivate, this.lblProtected, this.PictureBox1, this.btnClose, this.lblTitle});

        this.Dock = (System.Windows.Forms.DockStyle) resources.GetObject("$this.Dock");

        this.Enabled = (bool) resources.GetObject("$this.Enabled");

        this.Font = (System.Drawing.Font) resources.GetObject("$this.Font");

        this.Icon = (System.Drawing.Icon) resources.GetObject("$this.Icon");

        this.ImeMode = (System.Windows.Forms.ImeMode) resources.GetObject("$this.ImeMode");

        this.Location = (System.Drawing.Point) resources.GetObject("$this.Location");

        this.MaximizeBox = false;

        this.MaximumSize = (System.Drawing.Size) resources.GetObject("$this.MaximumSize");

        this.MinimumSize = (System.Drawing.Size) resources.GetObject("$this.MinimumSize");

        this.Name = "frmBase";

        this.RightToLeft = (System.Windows.Forms.RightToLeft) resources.GetObject("$this.RightToLeft");

        this.StartPosition = (System.Windows.Forms.FormStartPosition) resources.GetObject("$this.StartPosition");

        this.Text = resources.GetString("$this.Text");

        this.Visible = (bool) resources.GetObject("$this.Visible");

        this.ResumeLayout(false);

		base.Load +=new EventHandler(frmBase_Load);

    }

#endregion

    // Creates a connection to the database and uses a SqlDataAdapter
    // object to fill a DataSet.


    protected DataSet GetDataSource()
		{
        
        // Display a status message saying that we're attempting to connect.
        // This only needs to be done the very first time a connection is
        // attempted.  After we've determined that MSDE or SQL Server is
        // installed, this message no longer needs to be displayed.

        frmStatus frmStatusMessage = new frmStatus();
		DataSet dsProducts = null;

        if (!DidPreviouslyConnect) {

            frmStatusMessage.Show("Connecting to SQL Server");

        }

        // Attempt to connect to the local SQL server instance, and a local
        // MSDE installation (with Northwind).  

        bool IsConnecting = true;

        while (IsConnecting)
			{
            try {
                // The SqlConnection class allows you to communicate with SQL Server.
                // The constructor accepts a connection string an argument.  This
                // connection string uses Integrated Security, which means that you 
                // must have a login in SQL Server, or be part of the Administrators
                // group for this to work.

                SqlConnection cnNorthwind = new SqlConnection(connectionstring);

                // The SqlDataAdapter is used to move data between SQL Server, 
                // and a DataSet.

                SqlDataAdapter da = new SqlDataAdapter(
                    "SELECT ProductID, ProductName, UnitPrice, UnitsInStock " +
                    "FROM products", cnNorthwind);

                dsProducts = new DataSet();

                // Populate the Dataset with the information from the products
                // table.  Since a Dataset can hold multiple result sets, it's
                // a good idea to "name" the result set when you populate the
                // DataSet.  In this case, the result set is named "Products".

                da.Fill(dsProducts, "Products");

                // Data has been successfully retrieved, so break out of the loop
                // and close the status form.

                IsConnecting = false;
                DidPreviouslyConnect = true;
                frmStatusMessage.Close();

           } catch
			{
                if (connectionstring == SQL_CONNECTION_STRING) {

                    // Couldn't connect to SQL Server.  Now try MSDE.

                    connectionstring = MSDE_CONNECTION_STRING;
                    frmStatusMessage.Show("Connecting to MSDE");
					}
                else {

                    // Unable to connect to SQL Server or MSDE

                    frmStatusMessage.Close();
                    MessageBox.Show("To run this sample, you must have SQL " +
                        "or MSDE with the Northwind database installed.  For " + 
                        "instructions on installing MSDE, view the ReadMe file.",
                        "Connection Problem", MessageBoxButtons.OK, 
                        MessageBoxIcon.Error);
					Application.Exit();
                }
				
            }
			
        }
		
        frmStatusMessage.Close();
		return dsProducts;
    }

    private void frmBase_Load(object sender, System.EventArgs e)
	{

    }

}

